trickfs/src/lib.rs

```diff
@@ -0,0 +1,1021 @@
+use std::{
```

Comment:
This is a really cool piece of code, would be nice to have a Rustdoc here explaining what's being achieved and how - ofc I'm being selfish here as I'd like to have that for my own sake of learning.